src/Nancy/Responses/Negotiation/DefaultResponseNegotiator.cs

```diff
@@ -328,16 +328,18 @@ private static NegotiationContext GetNegotiationContext(object routeResult, Nanc
         /// </summary>
         /// <param name="requestUrl">The request URL.</param>
         /// <param name="linkProcessors">The link processors.</param>
+        /// <param name="existingLinkHeader">The existing Link HTTP Header.</param>
         /// <returns>The link header.</returns>
-        private static string CreateLinkHeader(Url requestUrl, IEnumerable<KeyValuePair<string, MediaRange>> linkProcessors)
+        protected virtual string CreateLinkHeader(Url requestUrl, IEnumerable<KeyValuePair<string, MediaRange>> linkProcessors, string existingLinkHeader)
         {
             var fileName = Path.GetFileNameWithoutExtension(requestUrl.Path);
             var baseUrl = string.Concat(requestUrl.BasePath, "/", fileName);
 
             var links = linkProcessors
-                .Select(lp => string.Format("<{0}.{1}>; rel=\"{2}\"", baseUrl, lp.Key, lp.Value));
+                .Select(lp => string.Format("<{0}.{1}>; rel=\"alternate\"; type=\"{2}\"", baseUrl, lp.Key, lp.Value))
+                .Concat(new[] { existingLinkHeader });

```

Comment:
Hmm. I just did a quick test:

```csharp
Console.WriteLine(string.Join(", ", new[] { "hello", "testing", null }));
```

Result: `"hello, testing, "`

That's not good. I think we need to check for `null` before concatenating here.